pzsd_bot/cogs/points.py

```diff
@@ -200,18 +222,32 @@ async def fetch_leaderboard(
                 .where(*args)
                 .group_by(pzsd_user.c.id)
             )
-            leaderboard = sorted(result.fetchall(), key=lambda r: r.sum, reverse=True)
+            sorted_points = sorted(result.fetchall(), key=lambda r: r.sum, reverse=True)
+
+        logger.info("Leaderboard length is %s", len(sorted_points))
+
+        leaderboard = [
```

Comment:
you can declare this as a generator rather than a list comprehension to save you from iterating over the list twice

`leaderboard = ((rank, name, points) for rank, (name, points) in enumerate(sorted_points, start=1))`